### PR TITLE
Noted APIs deprecated for new development

### PR DESCRIPTION
**Pull request by rpetrusha**
## Noted APIs deprecated for new development

Fixes dotnet/docs#917
Contributes to dotnet/docs#4121

//cc @terrajobst

### CODE REVIEW

**Comment by pkulikov** [comment]
@rpetrusha if not fixes (maybe even that), this PR contributes to dotnet/docs#4121

**Comment by rpetrusha** [comment]
Thanks for pointing out dotnet/docs#4121, @pkulikov. I'll add it, and take a close look at ArrayList to see if it needs additional revision.

File: xml/System.Collections/ArrayList.xml [hunk 1]
```diff
@@ -56,7 +56,11 @@
       <format type="text/markdown"><![CDATA[  
   
 ## Remarks  
- The <xref:System.Collections.ArrayList> class is designed to hold heterogeneous collections of objects. However, it does not always offer the best performance. Instead, we recommend the following:  
+
+>  [!IMPORTANT]
+>  We don't recommend that you use the `ArrayList` class for new development. Instead, we recommend that you use the generic <xref:System.Collections.Generic.List%601> class. For more information, see [Non-generic collections shouldn't be used](https://github.com/dotnet/platform-compat/blob/master/docs/DE0006.md) on GitHub.
+
+The <xref:System.Collections.ArrayList> class is designed to hold heterogeneous collections of objects. However, it does not always offer the best performance. Instead, we recommend the following:
```
Comment (svick) [inline xml/System.Collections/ArrayList.xml]: With this change, there is now a note recommending to use `List<T>`, followed by another recommendation to use `List<T>`. Should these two be combined? Or at least connected somehow?

Comment (rpetrusha) [reply xml/System.Collections/ArrayList.xml]: Yes, it should be, @svick. Thanks for pointing that out. I should have read the documentation for ArrayList more carefully before adding the note.

**Comment by imba-tjd** [comment]
About OSVersion.
I think the result of win8 or later should be kept a little bit, otherwise many people would have questions, although there are answers on StackOverflow. And actually editing manifest can get the right result on win10.
`System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform` accept `System.Runtime.InteropServices.OSPlatform`, which seems not accurate enough. It only has `Windows`, `OSX` and `Linux`.
`System.Runtime.InteropServices.RuntimeInformation.OSDescription` has more information, but I don't know OSVersion or OSDescription which one is more reliable on Linux and Mac OS.

**Pull request merged**